### PR TITLE
style: add % to total to area chart tooltip

### PR DESCRIPTION
**Pull request by thomaswang-shift**
This PR formats the tooltip for the area chart to include a new column for displaying `% to total`.

### CODE REVIEW

File: packages/superset-ui-legacy-preset-chart-nvd3/src/utils.js [hunk 1]
```diff
@@ -132,7 +132,29 @@ export function generateRichLineTooltipContent(d, timeFormatter, valueFormatter)
...
   tooltip += '</tbody></table>';
+  return tooltip;
+}
 
+export function generateAreaChartTooltipContent(d, timeFormatter, valueFormatter) {
+  const total = d.series[d.series.length - 1].value;
+  let tooltip = '';
+  tooltip +=
+    "<table><thead><tr><td colspan='3'>" +
+    `<strong class='x-value'>${timeFormatter(d.value)}</strong>` +
```
Comment (mistercrunch) [inline packages/superset-ui-legacy-preset-chart-nvd3/src/utils.js:144]: Should we sanitize / "html escape" what gets evaluated throughout this snipet?

**Comment by dorq** [comment]
From @datability-io ...
image (6)

**Comment by dorq** [comment]
This is looking great!  A few questions:
- Does this PR remove the colored blocks next to the values?  From another area chart in production now:
image
- Does this PR remove the bold and line styling on the hover value and the total?
- Do we need to include "category" and "value" as column headers?  If we do, can we capitalize them?  I'm assuming these names are standard across all charts.
- Can we left-align the date at the top with the first column below it?

File: packages/superset-ui-legacy-preset-chart-nvd3/src/utils.js (hunk 1 above)
Comment (xtinec) [reply packages/superset-ui-legacy-preset-chart-nvd3/src/utils.js:144]: Is this what you mean @mistercrunch ?
https://stackoverflow.com/questions/7381974/which-characters-need-to-be-escaped-in-html
I wonder if we have a utility function for this.

Comment (mistercrunch) [reply packages/superset-ui-legacy-preset-chart-nvd3/src/utils.js:144]: We use `dompurify.sanitize` in other places. Basically preventing XSS. In theory everything that is inputed by a user (webform) or comes from an external database should be sanitize when put into raw html... If `d.value` contains `<script>doBadStuff();</script>` it shouldn't get evaluated, it should just show up in the tooltip.
https://github.com/cure53/DOMPurify#what-does-it-do

Comment (mistercrunch) [reply packages/superset-ui-legacy-preset-chart-nvd3/src/utils.js:144]: A bit more about this. React "sanitizes" ("html escapes") all strings you put into it, everywhere by default. If you want user input to get rendered as HTML for instance, you'll have to use `dangerouslySetInnerHTML` https://zhenyong.github.io/react/tips/dangerously-set-inner-html.html

I think Jinja too, unless you mark something as safe.

So anytime we see html strings getting concatenated with user-provided content, we need to watch out for XSS.

Comment (thomaswang-shift) [reply packages/superset-ui-legacy-preset-chart-nvd3/src/utils.js:144]: Well explained! @mistercrunch

**Comment by kristw** [comment]
Please make sure your code pass travis CI

**Comment by thomaswang-shift** [comment]
This is the result of the latest change.  We should be good to go on this PR.

image

**Comment by thomaswang-shift** [comment]
@kristw, it looks like the Travis CI failure is not related to the change in this PR and it's also failing on master.

**Comment by dorq** [comment]
LGTM.  Thanks, @datability-io !

**Comment by kristw** [comment]
Your changes fails linting.

image

File: packages/superset-ui-legacy-preset-chart-nvd3/src/utils.js [hunk 2]
```diff
@@ -133,7 +133,36 @@ export function generateRichLineTooltipContent(d, timeFormatter, valueFormatter)
...
+  const total = d.series[d.series.length - 1].value;
+  let tooltip = '';
+  tooltip +=
+    "<table><thead><tr><td colspan='4'>" +
+    `<strong class='x-value'>${timeFormatter(d.value)}</strong>` +
+    '</td></tr></thead><tbody>' +
+    '<tr><td></td><td>Category</td><td>Value</td><td>% to total</td></tr>';
+  d.series.forEach(series => {
+    const key = getFormattedKey(series.key, true);
+    let trCls = '';
```
Comment (kristw) [inline packages/superset-ui-legacy-preset-chart-nvd3/src/utils.js]: Please use more specific variable name. Perhaps `trClass`

**Comment by kristw** [comment]
something is wrong. Please make sure you rebase from upstream master correctly.

**Comment by thomaswang-shift** [comment]
🔨

**Comment by kristw** [comment]
@datability-io This PR modifies 64 files. I don't think that is your intention. 
image

**Review by xtinec — APPROVED** [review]
Looks good!

**Review by kristw — APPROVED** [review]

**Pull request merged**